topics/RSA-Keys-Generation.md

```diff
@@ -0,0 +1,240 @@
+* `xxd -p -r`: This command is used to convert hexadecimal to binary. It takes the hexadecimal input and produces the corresponding binary output.
+* `| base64`: This part of the command takes the binary output from the previous step and encodes it in Base64 format using the `base64` command.
+
+<note>
+<p>
+Note that we used an even number of hex digits by padding an extra 0 to the left.
+</p>
+</note>
+</tab>
+</tabs>
+
+Let's see the output of this command for the aforementioned exponent value:
+
+```Shell
+$ echo 010001 | xxd -p -r | base64
+AQAB
+```
+
+The Base64URL encoded value of the exponent is `AQAB` and does not require further processing for this case.
+In other cases, you may need to use the `tr` command as we will see in the next step.
+
+### Modulus
+
+Let's perform the same process now, for the `n` attribute. This time, we will use the `tr` utility to
+further process the hexadecimal representation of the modulus.
+
+<tabs>
+<tab title="OpenSSL">
+<code-block lang="shell">
+echo "b5:f2:5a:2e:bc:d7:20:b5:20:d5:4d:cd:d4:a5:
+    7c:c8:9a:fd:d8:61:e7:e4:eb:58:65:1e:ea:5a:4d:
+    4c:73:87:32:e0:91:a3:92:56:2e:a7:bc:1e:32:30:
+    43:f5:fd:db:05:5a:08:b2:25:15:5f:ac:4d:71:82:
+    2b:d0:87:b4:01" | tr -d ": \n" | xxd -p -r | base64 | tr +/ -_ | tr -d "=\n"
+</code-block>
+
+* `echo "b5:f2:5a:2e:bc:d7:20:b5:20:d5:4d:cd:d4:a5: \ ... "`: This part of the command echoes a multi-line hexadecimal string, representing a series of bytes. The backslashes at the end of each line indicate line continuation.
+* `tr -d ": \n"`: The `tr` command is used to delete characters specified in the argument list.
+Here, it removes colons, spaces, and newline characters from the hexadecimal string, making it a continuous string of hex digits.
+* `xxd -p -r`: `xxd` is a utility for creating a hex dump of a binary file or converting a hex dump back to binary.
+The `-p` option specifies plain hexdump without line number or ASCII character columns.
+The `-r` option reverses the operation, converting hex back to binary.
+* `base64`: Encodes the binary output from the previous step into Base64 format.
+* `tr +/ -_`: Translates the + and / characters in the Base64 output to - and _, respectively. This is a common modification for URL-safe Base64 encoding.
+* `tr -d "=\n"`: Removes any equal signs (=) and newline characters from the final Base64-encoded string.
+</tab>
+</tabs>
+
+The output of the above command is:
+
+```Shell
+$ echo "b5:f2:5a:2e:bc:d7:20:b5:20:d5:4d:cd:d4:a5:
+    7c:c8:9a:fd:d8:61:e7:e4:eb:58:65:1e:ea:5a:4d:
+    4c:73:87:32:e0:91:a3:92:56:2e:a7:bc:1e:32:30:
+    43:f5:fd:db:05:5a:08:b2:25:15:5f:ac:4d:71:82:
+    2b:d0:87:b4:01" | tr -d ": \n" | xxd -p -r | base64 | tr +/ -_ | tr -d "=\n"
+tfJaLrzXILUg1U3N1KV8yJr92GHn5OtYZR7qWk1Mc4cy4JGjklYup7weMjBD9f3bBVoIsiUVX6xNcYIr0Ie0AQ
+```
+
+<note>
+<p>
+Note that the leading 00 byte has been omitted; that's an artifact of the ASN.1 encoding.
```

Comment:
I don't think it's clear to everyone what this "leading 00 byte" is here. Perhaps you can explain it a bit more?

Reply:
I added an explanation based on what I could find online with a quick read. If it's not correct or enough, let me know.